disease_gene/disease_associates_gene/generative_model_experiments/data/label_functions/disease_gene_lf.py

```diff
@@ -0,0 +1,812 @@
+
+
+def LF_DG_CHECK_GENE_TAG(c):
+    """
+    This label function is used for labeling each passed candidate as either pos or neg.
+    Keyword Args:
+    c- the candidate object to be passed in.
+    """
+    sen = c[1].get_parent()
+    gene_name = re.sub("\)", "", c[1].get_span().lower())
+    gene_id = sen.entity_cids[c[1].get_word_start()]
+    gene_entry_df = gene_desc.query("GeneID == @gene_id")
+
+    if gene_entry_df.empty:
+        return -1
+
+    for token in gene_name.split(" "):
+        if gene_entry_df["Symbol"].values[0].lower() == token or token in gene_entry_df["Synonyms"].values[0].lower():
+            return 0
+        elif token in gene_entry_df["description"].values[0].lower():
+            return 0
+    return -1
+
+
+#disease_desc = pd.read_table("https://raw.githubusercontent.com/dhimmel/disease-ontology/052ffcc960f5897a0575f5feff904ca84b7d2c1d/data/xrefs-prop-slim.tsv")
+disease_normalization_df = pd.read_table("https://raw.githubusercontent.com/dhimmel/disease-ontology/052ffcc960f5897a0575f5feff904ca84b7d2c1d/data/slim-terms-prop.tsv")
+wordnet_lemmatizer = WordNetLemmatizer()
+
+def LF_DG_CHECK_DISEASE_TAG(c):
+    """
+    This label function is used for labeling each passed candidate as either pos or neg.
+    Keyword Args:
+    c- the candidate object to be passed in.
+    """
+    sen = c[0].get_parent()
+    disease_name = re.sub("\) ?", "", c[0].get_span())
+    disease_name = re.sub(r"(\w)-(\w)", r"\g<1> \g<2>", disease_name)
+    disease_name = " ".join([word for word in word_tokenize(disease_name) if word not in set(stop_word_list)])
+
+    # If abbreviation skip since no means of easy resolution
+    if len(disease_name) <=5 and disease_name.isupper():
+        return 0
+
+    disease_id = sen.entity_cids[c[0].get_word_start()]
+    result = disease_normalization_df[
+        disease_normalization_df["subsumed_name"].str.contains(disease_name.lower(), regex=False)
+    ]
+    # If no match then return -1
+    if result.empty:
+
+        # check the reverse direction e.g. carcinoma lung -> lung carcinoma
+        disease_name_tokens = word_tokenize(disease_name)
+        if len(disease_name_tokens) == 2:
+            result = disease_normalization_df[
+                disease_normalization_df["subsumed_name"].str.contains(" ".join(disease_name_tokens[-1::0-1]).lower(), regex=False)
+            ]
+            
+            # if reversing doesn't work then output -t
+            if not result.empty:
+                slim_id = result['slim_id'].values[0]
+                if slim_id == disease_id:
+                    return 0
+        return -1
+    else:
+        # If it can be normalized return 0 else -1
+        slim_id = result['slim_id'].values[0]
+        if slim_id == disease_id:
+            return 0
+        else:
+            return -1
+
+"""
+SENTENCE PATTERN MATCHING
+"""
```

Comment:
These sentence motifs were generated by you based on reading?  They were not pulled from some ontology?  I would add the source to your comment